crates/sui-graphql-rpc/src/types/transaction_block.rs

```diff
@@ -596,3 +798,31 @@ impl TryFrom<TransactionBlockEffects> for TransactionBlock {
         })
     }
 }
+
+// fn max_option(a: Option<u64>, b: Option<u64>) -> Option<u64> {
```

Comment:
I would recommend using a CTE and keeping this logic written in SQL, to avoid roundtrips where we can.